projects/core/directives/dropdown/dropdown-position.directive.ts

```diff
@@ -44,15 +44,17 @@ export class TuiDropdownPositionDirective extends TuiPositionAccessor {
             top: hostRect.top - 2 * offset - viewport.top,
             bottom: viewport.bottom - hostRect.bottom - 2 * offset,
         } as const;
+        const left =
+            hostRect.left + width < viewport.right - offset ? hostRect.left : right;
```

Comment:
I wonder if we can somehow simplify all the offset subtractions by just subtracting it from the viewport rect initially 🤔